include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -369,39 +387,35 @@ inline void multiplexer::do_start_receive()
 {
     auto self(shared_from_this());
 
-    // We need to read with at least one zero sized buffer to
-    // get the remote_endpoint information.
-    next_layer().async_receive_from
-        (boost::asio::buffer(static_cast<char*>(nullptr), 0),
-         next_remote_endpoint,
-         std::remove_reference<decltype(next_layer())>::type::message_peek,
+    next_layer().async_receive
+        (boost::asio::null_buffers(),
          [self]
          (boost::system::error_code error, std::size_t /*size*/) mutable
          {
-            // The size parameter is useless here because what we get
-            // is min(buffer_size, datagram_size) and our buffer size is 0.
-            self->process_peek(error, self->next_remote_endpoint);
+            self->process_new_message(error);
```

Comment:
@inetic null_buffers is simply type sugar to tell ASIO to not actually do any reading, just do the part which sets up the async notification of a non-blocking read being available. null_buffers is specifically guaranteed to not consume UDP packets. See https://stackoverflow.com/questions/20419704/variable-size-buffer-for-receiving-udp-packets for an example.

I didn't do the async_peek(header) because you said you wanted the empty message to be as small as possible, which I defined to be 2 bytes. If you don't mind empty message being 12 bytes, then no problem we can async peek the header, though note that as a consequence you're doing an extra malloc per async peek before we've filtered out the DDoS traffic which may or may not be important. Let me know which you prefer, I still have an unit parameter warning/error to fix anyway.